### PR TITLE
ADNI VIT (Task 6)

### PR DESCRIPTION
**Pull request by electricalengi**
This project implemented a standard VIT for classifying the ADNI Alzheimer's data. The model achieved above 80% validation accuracy and above 60% test accuracy.

Callum Higgins, 46410401, Task 6

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
Great work.

Model design: Great.
Data split: missing a separate test set. You will need a test set to test your model performance after training. Currently, you only have a validation set which you call "test" in your assignment.
README: Clearly explained
Commit logs: Great
Training process: Correct

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage, but no test set -2
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description OK but could be more informative -1

**Comment by electricalengi** [comment]
> 
Hi Shakes,
I was just wondering if this counts as a test set?
`def get_test_data():
    AUTOTUNE = tf.data.AUTOTUNE
    test_data = keras.utils.image_dataset_from_directory(
        directory='test/',
        labels='inferred',
        label_mode='int',
        batch_size=32,
        image_size=(256, 240))

    test_data = test_data.cache().shuffle(1000).prefetch(buffer_size=AUTOTUNE)
    return test_data`

In dataset.py

**Comment by electricalengi** [comment]
Hi shakes, there is a test set added in commit "Added VIT model creation" in dataset.py
